flutter/lib/backend/loadgen_info.dart

```diff
@@ -0,0 +1,82 @@
+import 'dart:convert';
+import 'dart:io';
+
+class LoadgenInfo {
+  // Mean latency in seconds
+  final double meanLatency;
+  final int queryCount;
+  // 90th percentile in seconds
+  final double latency90;
+  final bool validity;
+
+  LoadgenInfo({
+    required this.meanLatency,
+    required this.queryCount,
+    required this.latency90,
+    required this.validity,
+  });
+
+  static Future<LoadgenInfo?> fromFile({required String filepath}) {
+    final lines = File(filepath)
+        .openRead()
+        .map(utf8.decode)
+        .transform(const LineSplitter());
+    return extractLoadgenInfo(logLines: lines);
+  }
+
+  static Future<Map<String, dynamic>> extractKeys({
+    required Stream<String> logLines,
+    required Set<String> requiredKeys,
+  }) async {
+    Map<String, dynamic> result = {};
+
+    await for (var line in logLines) {
+      const prefix = ':::MLLOG ';
+      if (!line.startsWith(prefix)) {
+        throw 'missing $prefix prefix';
+      }
+      line = line.substring(prefix.length);
+      final json = jsonDecode(line);
+      final lineKey = json['key'] as String;
+      if (requiredKeys.contains(lineKey)) {
+        result[lineKey] = json['value'];
+      }
+    }
+
+    return result;
+  }
+
+  static Future<LoadgenInfo?> extractLoadgenInfo({
+    required Stream<String> logLines,
+  }) async {
+    const latencyKey = 'result_mean_latency_ns';
+    const queryCountKey = 'result_query_count';
+    const latency90Key = 'result_90.00_percentile_latency_ns';
+    const validityKey = 'result_validity';
+
+    final result = await extractKeys(
+      logLines: logLines,
+      requiredKeys: {
+        latencyKey,
+        queryCountKey,
+        latency90Key,
+        validityKey,
+      },
+    );
+
+    if (result.isEmpty) {
+      return null;
+    }
+
+    final validity = result[validityKey] as String == 'VALID';
+
+    const nanosecondsPerSecond = 1000 * Duration.microsecondsPerSecond;
+
+    return LoadgenInfo(
+      meanLatency: (result[latencyKey] as int) / nanosecondsPerSecond,
+      queryCount: result[queryCountKey] as int,
+      latency90: (result[latency90Key] as int) / nanosecondsPerSecond,
+      validity: validity,
+    );
```

Comment:
Why do we need perfect precision for this? I don't see a point in complicating things.

Reply:
The latency is used for other calculations. I don't know yet if we need that precision, but I also don't see the need to convert it to second either, since second is certainly not precise enough in the context of this app (`benchmarking`), so I prefer to just using the original value.

Reply:
For duration calculation, we can use the built-in class [Duration](https://api.dart.dev/stable/1.24.2/dart-core/Duration-class.html) when we need to convert between units. So we can either: 
1. keep the original precision of nanosecond here and use the `Duration` class later when needed. Or
2. use the `Duration` class here with the knowledge that `Duration` only support up to microsecond precision. But microsecond should be enough, I think. Second is not enough precision for this app IMHO.

Reply:
>since second is certainly not precise enough

Seconds are almost as precise as nanoseconds, because we use `double` datatype.

>we can use the built-in class [Duration](https://api.dart.dev/stable/1.24.2/dart-core/Duration-class.html)

By the way, `Duration` doesn't support nanoseconds, and its other methods for milli- and microseconds aren't as precise as using plain `double` values (of any order).

>The latency is used for other calculations

All _really_ important calculations are already done in the loadgen. I mean, those calculations, for which even nanoseconds in floating point datatype would not be enough.
For calculations in the UI that possible `0.0001%` error doesn't matter.